subprojects/composite-builds/src/main/java/org/gradle/composite/internal/IncludedBuildDependencyMetadataBuilder.java

```diff
@@ -77,8 +77,8 @@ private LocalComponentMetadata createCompositeCopy(IncludedBuild build, ProjectC
             compositeComponentMetadata.addConfiguration(configurationName,
                 originalConfiguration.getDescription(), originalConfiguration.getExtendsFrom(), originalConfiguration.getHierarchy(),
                 originalConfiguration.isVisible(), originalConfiguration.isTransitive(), originalConfiguration.getAttributes(),
-                originalConfiguration.isConsumeOrPublishAllowed(),
-                originalConfiguration.isQueryOrResolveAllowed());
+                originalConfiguration.isCanBeConsumed(),
```

Comment:
Yes and no. `canBeConsumed` conveys the concept of access right:

> Can I consume this, please?

Whereas `consumable` conveys the idea of _state_:

> Is this sushi consumable?

Which is why I gave up and went with `canXXX` as ugly as it can be (pun intended). I'm not sure we spend more time on picking better names now, they already changed at least 5 times during this review and nobody was ever happy :)